### PR TITLE
only set upid.Host and upid.Port if Host is unset

### PR DESCRIPTION
**Pull request by antonlindstrom**
This commit checks if `upid.Host` is set in the http_transporter and sets it
to the value that it determines if unset.

The original problem was that I was trying to run a scheduler on 0.0.0.0 as
listening host and have OverrideHostname set, this should have led to the
master thinking that I was sending messages from HostnameOverride, instead the
http_transporter function listen(), mutated the upid.Host and the master would
always look up 0.0.0.0.

The patch has been tested by running a scheduler against a Mesos cluster with
the scheduler bound to 0.0.0.0 and HostnameOverride in DriverConfig to an
external IP.

Thanks!

### CODE REVIEW

**Comment by jdef** [comment]
Thanks for the PR! Can you also write up a quick unit test?

**Comment by antonlindstrom** [comment]
@jdef Thanks! I added a simple test that covered my case, it's fairly simple though.

File: messenger/http_transporter_test.go [hunk 1]
```diff
@@ -266,6 +267,34 @@ func TestTransporterStartAndStop(t *testing.T) {
...
+
+	// override the upid.Host with this listener IP
+	addr := net.ParseIP("127.0.1.1")
+
+	// setup receiver (server) process
+	uPid, err := upid.Parse(fmt.Sprintf("%s@%s", serverId, serverAddr))
+	assert.NoError(t, err)
+	receiver := NewHTTPTransporter(uPid, addr)
+
+	errch := receiver.Start()
```
Comment (jdef) [inline messenger/http_transporter_test.go]: - why call Start/Stop instead of listen(), which is the func that you really changed?
- this is great for testing the code as it was *before* your changes. how about adding a test case that stresses the code you added to HTTPTransporter?

File: messenger/http_transporter_test.go [hunk 2]
```diff
@@ -266,6 +267,34 @@ func TestTransporterStartAndStop(t *testing.T) {
 	}
 }
 
+func TestMutadedHostUPid(t *testing.T) {
```
Comment (jdef) [inline messenger/http_transporter_test.go]: s/Mutaded/Mutated

**Comment by antonlindstrom** [comment]
@jdef Thanks for the feedback! I've updated the tests to include a test where the upid host and port are empty. I think that should cover what I actually changed.

Thanks!

File: messenger/http_transporter_test.go [hunk 3]
```diff
@@ -266,6 +267,66 @@ func TestTransporterStartAndStop(t *testing.T) {
...
+
+	// override the upid.Host with this listener IP
+	addr := net.ParseIP("127.0.1.1")
+
+	receiver := NewHTTPTransporter(uPid, addr)
+
+	err = receiver.listen()
+	assert.NoError(t, err)
+
+	// This should be the port that overrides as uPid.Host is empty
```
Comment (jdef) [inline messenger/http_transporter_test.go]: s/port/host

**Comment by jdef** [comment]
Minor doc error. Otherwise LGTM

**Comment by antonlindstrom** [comment]
Updated, thanks!
